pkg/operation/botanist/health_check.go

```diff
@@ -306,39 +308,59 @@ func (b *HealthChecker) CheckSystemComponents(
 	if err != nil {
 		return nil, err
 	}
-	if exitCondition := b.checkRequiredDeployments(condition, common.RequiredSystemComponentDeployments, deploymentList); exitCondition != nil {
+
+	var (
+		requiredSystemComponentDeployments = sets.NewString(
+			common.CoreDNSDeploymentName,
+			common.MetricsServerDeploymentName,
+		)
+
+		requiredSystemComponentDaemonSets = sets.NewString(
+			common.KubeProxyDaemonSetName,
+			common.NodeProblemDetectorDaemonSetName,
+		)
+	)
```

Comment:
Why are they moved here? I would rather keep them in `common`, make a copy here and add either VPN/Konnectivity

Reply:
Because when they were global, having multiple shoots reconcile and Chang their value would lead to wrong outputs (e.g shoot with VPN would add an undesirable requirement)

Reply:
> make a copy here

I have no strong opinion here, the only that concerns me is that now it's inconsistent (some of these "Required..." lists are still in `common`, but now some are also only "privately" here). I'd try to keep this harmonized.